ballerina/tests/custom_remote_functions.bal

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023 WSO2 LLC. (//www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// //www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/io;
+
+listener Listener customDispatchingLis = new(21401);
+
+@ServiceConfig {dispatcherKey: "event"}
+service on customDispatchingLis {
+    resource function get .() returns Service|Error {
+        return new SingleMessageService();
+    }
+}
+
+service class SingleMessageService {
+    *Service;
+    remote function onPing(Caller caller, json data) returns Error? {
+        io:println(data);
+        check caller->writeMessage({"event": "pong"});
+    }
+}
+
+@ServiceConfig {dispatcherKey: "event"}
+service /subscribe on customDispatchingLis {
+    resource function get .() returns Service|Error {
+        return new SubscribeMessageService();
+    }
+}
+
+service class SubscribeMessageService {
+    *Service;
+    remote function onSubscribe(Caller caller, json data) returns Error? {
+        io:println(data);
+        check caller->writeMessage({"type": "subscribe", "id":"1", "payload":{"query": "{ __schema { types { name } } }"}});
+    }
+
+    remote function onHeartbeat(Caller caller, json data) returns json {
+        return {"event": "heartbeat"};
```

Comment:
Shall we also add a handler for an event name having `_` ? ex: {"event": "on_ping"};

Reply:
Did you mean what is covered by these 2 test cases? or is it something else?
https://github.com/ballerina-platform/module-ballerina-websocket/pull/900/files#diff-255c8647fe340cb8fb39a6f681b5cf7d2cc035820245ad7cddb18d094e9473e1R175-R188

Reply:
Yes. Seems it's covered